nux/components/dot-tip/index.js

```diff
@@ -0,0 +1,95 @@
+/**
+ * External dependencies
+ */
+import { defer } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { Component, createRef, compose } from '@wordpress/element';
+import { Popover, Button, IconButton } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { withSelect, withDispatch } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+
+export class DotTip extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.popoverRef = createRef();
+	}
+
+	componentDidMount() {
+		if ( this.props.isVisible ) {
+			// Force the popover to recalculate its position on the next frame. This is a
+			// Temporary workaround to fix the tip not appearing next to the inserter
+			// toggle on page load. This happens because the popover calculates its
+			// position before <PostTitle> is made visible, resulting in the position
+			// being too high on the page.
+			defer( () => {
+				const popover = this.popoverRef.current;
+				const popoverSize = popover.updatePopoverSize();
+				popover.computePopoverPosition( popoverSize );
+				popover.focus();
+			} );
+		}
+	}
+
+	render() {
+		const { children, isVisible, hasNextTip, onDismiss } = this.props;
+
+		if ( ! isVisible ) {
+			return null;
+		}
+
+		return (
+			<Popover
+				ref={ this.popoverRef }
+				className="nux-dot-tip"
+				position="middle right"
+				noArrow
+				focusOnMount
+				role="dialog"
+				aria-label={ __( 'Gutenberg tips' ) }
+				onClose={ onDismiss }
+				onClick={ ( event ) => event.stopPropagation() }
+			>
+				<p>{ children }</p>
+				<p>
+					<Button isLink onClick={ onDismiss }>
```

Comment:
```
Warning: validateDOMNesting(...): <button> cannot appear as a descendant of <button>.
    in button
    in p (created by DotTip)
```

Reply:
When do you see that warning, @aduth?

Reply:
I think it's with the default block appender `Inserter` tip, whose children are rendered within an `IconButton`:

https://github.com/WordPress/gutenberg/blob/c6cbfb041cf8c7860cbeaf1c36273d068e5e839e/editor/components/default-block-appender/index.js#L60-L64

https://github.com/WordPress/gutenberg/blob/c6cbfb041cf8c7860cbeaf1c36273d068e5e839e/editor/components/inserter/index.js#L60-L70